scaaml/capture/scope/ps6424e.py

```diff
@@ -163,21 +172,45 @@ def channel(self):
     @channel.setter
     def channel(self, val):
         if val not in self._ch_list:
-            raise ValueError("Unknown channel")
+            raise ValueError(f"Unknown channel {val} not in {self._ch_list}")
         self._channel = self._ch_list[val]
 
+        if self.is_digital:
+            if self._port_pin is None:
+                # Provide a sensible default if there is no value.
+                self._port_pin = 0
+        else:
+            # Analog channels do not have a pin.
+            self._port_pin = None
+
+    @property
+    def port_pin(self) -> Optional[int]:
+        """Number of pin of the digital port."""
+        return self._port_pin
+
+    @port_pin.setter
+    def port_pin(self, val: Optional[int]) -> None:
+        # If the channel is digital port is not None.
+        if self.is_digital:
+            assert val is not None
```

Comment:
I would keep assertions for validating boundaries within our code. Here it takes user input so I think raising ValueError would give a more actionnable error to the user.